include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -8463,119 +8271,44 @@ FEFaceEvaluation<dim,
     ExcMessage(
       "Only EvaluationFlags::values, EvaluationFlags::gradients, and EvaluationFlags::hessians are supported."));
 
-  const auto fu = [&]() {
-    const auto shared_vector_data = internal::get_shared_vector_data(
-      input_vector,
-      this->dof_access_index ==
-        internal::MatrixFreeFunctions::DoFInfo::dof_access_cell,
-      this->active_fe_index,
-      this->dof_info);
-
-    if (this->dof_access_index ==
-          internal::MatrixFreeFunctions::DoFInfo::dof_access_cell &&
-        this->is_interior_face == false)
-      {
-        const auto cells             = this->get_cell_or_face_ids();
-        const auto face_nos          = this->compute_face_no_data();
-        const auto face_orientations = this->compute_face_orientations();
+  const auto shared_vector_data = internal::get_shared_vector_data(
+    input_vector,
+    this->dof_access_index ==
+      internal::MatrixFreeFunctions::DoFInfo::dof_access_cell,
+    this->active_fe_index,
+    this->dof_info);
 
-        return internal::FEFaceEvaluationImplGatherEvaluateSelector<
+  if (fe_degree > 0 &&
+      internal::FEFaceEvaluationImplGatherEvaluateSelector<
+        dim,
+        Number,
+        VectorizedArrayType>::supports(evaluation_flag,
```

Comment:
What function is this?

Reply:
This is the new function to avoid an overly complicated choice of early returns and re-implementations. Now the case whether the `gather_evaluate`/`integrate_scatter` succeed is checked once up front and in case it is not, we do the conservative `read_dof_values` + `evaluate` calls. No hidden magic.